includes/class-wc-payments-fraud-service.php

```diff
@@ -203,4 +215,160 @@ public function add_sift_js_tracker() {
+				return WC_Payments_Utils::array_map_recursive(
+					$fraud_services,
+					function( $value ) {
+						// Only apply `sanitize_text_field()` to string values since this function will cast to string.
+						if ( is_string( $value ) ) {
+							return sanitize_text_field( $value );
+						}
+
+						return $value;
+					}
+				);
+			},
+			[ $this, 'is_valid_cached_fraud_services' ]
+		);
+	}
+
+	/**
+	 * Prepares the fraud config for a service.
+	 *
+	 * @param array  $config     Existing config data for the given anti-fraud service.
+	 * @param string $service_id Identifier of the anti-fraud service provider.
+	 *
+	 * @return array|NULL Array with all the required data to initialize the anti-fraud script, or NULL if the service shouldn't be used.
+	 */
+	private function prepare_fraud_config( array $config, string $service_id ): ?array {
+		switch ( $service_id ) {
+			case 'sift':
+				return $this->prepare_sift_config( $config );
+		}
+
+		return $config;
+	}
+
+	/**
+	 * Adds site-specific config needed to initialize the SIFT anti-fraud JS.
+	 *
+	 * @param array $config Associative array with the SIFT-related configuration returned from the server.
+	 *
+	 * @return array|null Assoc array, ready for the client to consume, or NULL if the client shouldn't enqueue this script.
+	 */
+	private function prepare_sift_config( array $config ): ?array {
+		$test_mode = false;
+		try {
+			$test_mode = WC_Payments::mode()->is_test();
+		} catch ( Exception $e ) {
+			Logger::log( sprintf( 'WooPayments JS settings: Could not determine if WCPay should be in test mode! Message: %s', $e->getMessage() ), 'warning' );
+		}
+
+		// The server returns both production and sandbox beacon keys. Use the sandbox one if test mode is enabled.
+		if ( $test_mode && isset( $config['sandbox_beacon_key'] ) ) {
+			$config['beacon_key'] = $config['sandbox_beacon_key'];
+		}
+		unset( $config['sandbox_beacon_key'] );
+
+		$config['user_id']    = $this->get_sift_user_id();
+		$config['session_id'] = $this->session_service->get_sift_session_id();
+
+		return $config;
+	}
+
+	/**
+	 * Get the Sift user ID to use for the current request.
+	 *
+	 * @return string The Sift user ID to use for the current request. Empty string if we couldn't determine one.
+	 */
+	private function get_sift_user_id(): string {
+		$user_id = '';
+
+		if ( ! is_user_logged_in() ) {
+			return $user_id;
+		}
+
+		if ( is_admin() ) {
+			// In the WP admin we deal with merchant accounts, not customers.
+			$user_id = $this->account->get_stripe_account_id() ?? '';
+		} else {
+			$customer_id = $this->customer_service->get_customer_id_by_user_id( get_current_user_id() );
+			if ( isset( $customer_id ) ) {
+				$user_id = $customer_id;
+			}
+		}
+
+		return $user_id;
+	}
+
+	/**
+	 * Fetches public fraud services config from the server.
+	 *
+	 * @return array|null Fraud services config or null.
+	 */
+	private function fetch_public_fraud_services_config(): ?array {
```

Comment:
It looks like we have the concept of [server request](https://github.com/Automattic/woocommerce-payments/tree/develop/includes/core/server) objects in the client available to us. What do you think about using that pattern here to fetch the fraud config from the server here?

Reply:
That would have been ideal, but how these abstractions are setup to be used is they are funneled through the client API that enforces the presence of a Jetpack connection. That is a no-go for merchants that haven't setup a Jetpack connection yet.